### PR TITLE
Labs

### PR DESCRIPTION
**Pull request by lizsyeo**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Completing lab1-6.

## What did you learn from the changes you have made?
Implementing neural network.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Yes, trying different approaches in training the model.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Yes, low memory and GPU issues.

## How were these changes tested?
Validation accuracy and other metrics.

## Checklist
- [x ] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by kunzhi-yu — APPROVED** [review]
All labs present.
